mailpoet/lib/Automation/Engine/Data/WorkflowRunLog.php

```diff
@@ -0,0 +1,174 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Engine\Data;
+
+use DateTimeImmutable;
+use InvalidArgumentException;
+use MailPoet\Automation\Engine\Exceptions\InvalidStateException;
+use MailPoet\Automation\Engine\Exceptions\UnexpectedValueException;
+use MailPoet\Automation\Engine\Utils\Json;
+use Throwable;
+
+class WorkflowRunLog {
+
+  const STATUS_RUNNING = 'running';
+  const STATUS_COMPLETED = 'completed';
+  const STATUS_FAILED = 'failed';
+
+  /** @var int */
+  private $id;
+
+  /** @var int */
+  private $workflowRunId;
+
+  /** @var DateTimeImmutable */
+  private $startedAt;
+
+  /** @var DateTimeImmutable|null */
+  private $completedAt;
+
+  /** @var string */
+  private $status;
+
+  /** @var array */
+  private $error;
+
+  /** @var array */
+  private $data;
+
+  /** @var string */
+  private $stepId;
+
+  /** @var array */
+  private $args;
```

Comment:
I wasn't sure if it would be meaningful to store the $args as part of the log or if we would be able to easily/reliably retrieve them from a related record, so I erred on the side of redundancy. Please let me know if you think we should remove this.

Reply:
These are the step args? I guess they're not run-specific, only workflow-specific. And since we have versioning, we could find them through that, if needed (log -> workflow run -> workflow version). I'm not sure if we'll need them in the log, what do you think?

Reply:
Yes, these are the step args. I should have made that clearer! I think that I'll remove them for now. The main concern would be if the step args could change before the step is run, or somehow get out of sync with what's in the workflow version, and that doesn't seem to be the case.

I also don't want to include more than what we know we'll need. It's not like it would be difficult to add step data back later if we come up with a specific use case for it. Let's keep is as simple as we can for now.